Sources/Tests/CoreTests/test_hsDarwin_NS.mm

```diff
@@ -47,21 +47,55 @@
 
 TEST(hsDarwin_Foundation, converts_to_ST_string)
 {
-    NSString* str = @"Test ö";
+    NSAutoreleasePool* pool = [[NSAutoreleasePool alloc] init];
+
+    NSString* str = @"Test 123";
     ST::string st = STStringFromNSString(str);
-    EXPECT_STREQ(st.c_str(), "Test \xc3\xb6");
+    EXPECT_STREQ(st.c_str(), "Test 123");
+
+    [pool drain];
 }
 
 TEST(hsDarwin_Foundation, converts_to_NSString)
 {
+    NSAutoreleasePool* pool = [[NSAutoreleasePool alloc] init];
+
+    ST::string st = ST_LITERAL("Test 123");
+    NSString* nstr = NSStringCreateWithSTString(st);
+    EXPECT_EQ(YES, [nstr isEqualToString:@"Test 123"]);
+
+    [pool drain];
+}
+
+TEST(hsDarwin_Foundation, converts_to_ST_string_UTF8)
+{
+    NSAutoreleasePool* pool = [[NSAutoreleasePool alloc] init];
+
+    NSString* str = [NSString stringWithUTF8String:"Test ö"];
+    ST::string st = STStringFromNSString(str);
+    EXPECT_STREQ(st.c_str(), "Test \xc3\xb6");
+
+    [pool drain];
+}
+
+TEST(hsDarwin_Foundation, converts_to_NSString_UTF8)
+{
+    NSAutoreleasePool* pool = [[NSAutoreleasePool alloc] init];
+
     ST::string st = ST_LITERAL("Test ö");
     NSString* nstr = NSStringCreateWithSTString(st);
-    EXPECT_EQ(YES, [nstr isEqualToString:@"Test ö"]);
+    EXPECT_EQ(YES, [nstr isEqualToString:[NSString stringWithUTF8String:"Test ö"]]);
+
+    [pool drain];
 }
 
 TEST(hsDarwin_Foundation, returns_retained_NSString)
 {
+    NSAutoreleasePool* pool = [[NSAutoreleasePool alloc] init];
+
     ST::string st = ST::format("{} Test {}", 12345, "Hello");
     NSString* nstr = NSStringCreateWithSTString(st);
     EXPECT_EQ(1, [nstr retainCount]);
+
+    [pool drain];
```

Comment:
I'm assuming this was not built with ARC on modern macOS either making this a leak?